migration-engine/connectors/sql-migration-connector/src/sql_schema_differ.rs

```diff
@@ -240,23 +231,36 @@ impl<'schema> SqlSchemaDiffer<'schema> {
         })
     }
 
-    fn drop_foreign_keys<'a>(differ: &'a TableDiffer<'schema>) -> impl Iterator<Item = TableChange> + 'a {
+    fn drop_foreign_keys<'a>(&'a self, drop_foreign_keys: &mut Vec<DropForeignKey>) {
+        for differ in self.table_pairs() {
+            let table_name = differ.previous.name();
+            for dropped_foreign_key_name in differ
+                .dropped_foreign_keys()
+                .filter_map(|foreign_key| foreign_key.constraint_name())
+            {
+                drop_foreign_keys.push(DropForeignKey {
+                    table: table_name.to_owned(),
+                    constraint_name: dropped_foreign_key_name.to_owned(),
+                })
+            }
+        }
+    }
+
+    fn add_primary_key(differ: &TableDiffer<'_>) -> Option<TableChange> {
         differ
-            .dropped_foreign_keys()
-            .filter_map(|foreign_key| foreign_key.constraint_name.as_ref())
-            .map(move |dropped_foreign_key_name| {
-                debug!(
-                    "Dropping foreign key '{}' on table '{}'",
-                    &dropped_foreign_key_name,
-                    &differ.previous.name()
-                );
-                let drop_step = DropForeignKey {
-                    constraint_name: dropped_foreign_key_name.clone(),
-                };
-                TableChange::DropForeignKey(drop_step)
+            .created_primary_key()
+            .filter(|pk| !pk.columns.is_empty())
+            .map(|pk| TableChange::AddPrimaryKey {
+                columns: dbg!(pk).columns.clone(),
```

Comment:
Was this `dbg` left in intentionally?

```suggestion
                columns: pk.columns.clone(),
```

Reply:
nope, good catch!